internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -262,6 +285,24 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 				}, false),
 		},
 
+		"web_application_firewall_settings": {
```

Comment:
settings is redundant, so we can remove it to also be more consistent with the rest of the provider
```suggestion
		"web_application_firewall": {
```

Reply:
@katbyte I have updated `web_application_firewall_settings` to `web_application_firewall` and nested the status inside of the `web_application_firewall` block instead of a separate block